src/Gizmos/planeDragGizmo.ts

```diff
@@ -0,0 +1,173 @@
+import { AbstractMesh } from "../Meshes/abstractMesh";
+import { Mesh } from "../Meshes/mesh";
+import { LinesMesh } from "../Meshes/linesMesh";
+import { PlaneBuilder } from "../Meshes/Builders/planeBuilder";
+import { PointerDragBehavior } from "../Behaviors/Meshes/pointerDragBehavior";
+import { _TimeToken } from "../Instrumentation/timeToken";
+import { _DepthCullingState, _StencilState, _AlphaState } from "../States/index";
+import { Gizmo } from "./gizmo";
+import { UtilityLayerRenderer } from "../Rendering/utilityLayerRenderer";
+import { StandardMaterial } from "../Materials/standardMaterial";
+import { Scene } from "../scene";
+import { PositionGizmo } from "./positionGizmo";
+/**
+ * Single plane drag gizmo
+ */
+export class PlaneDragGizmo extends Gizmo {
+    /**
+     * Drag behavior responsible for the gizmos dragging interactions
+     */
+    public dragBehavior: PointerDragBehavior;
+    private _pointerObserver: Nullable<Observer<PointerInfo>> = null;
+    /**
+     * Drag distance in babylon units that the gizmo will snap to when dragged (Default: 0)
+     */
+    public snapDistance = 0;
+    /**
+     * Event that fires each time the gizmo snaps to a new location.
+     * * snapDistance is the the change in distance
+     */
+    public onSnapObservable = new Observable<{ snapDistance: number }>();
+
+    private _isEnabled: boolean = false;
+    private _parent: PositionGizmo ;
+
+    /** @hidden */
+    public static _CreatePlane(scene: Scene, material: StandardMaterial): TransformNode {
+        var plane = new TransformNode("plane", scene);
+
+        //make sure plane is double sided
+        var dragPlane = PlaneBuilder.CreatePlane("dragPlane", { width: .1375, height: .1375, sideOrientation: 2 }, scene);
+        dragPlane.material = material;
+        dragPlane.parent = plane;
+
+        // Position plane pointing normal to dragPlane normal
+        dragPlane.material = material;
+        return plane;
+    }
+
+    /** @hidden */
+    public static _CreateArrowInstance(scene: Scene, arrow: TransformNode): TransformNode {
+        const instance = new TransformNode("arrow", scene);
+        for (const mesh of arrow.getChildMeshes()) {
+            const childInstance = (mesh as Mesh).createInstance(mesh.name);
+            childInstance.parent = instance;
+        }
+        return instance;
+    }
+
+    /**
+     * Creates a PlaneDragGizmo
+     * @param gizmoLayer The utility layer the gizmo will be added to
+     * @param dragPlaneNormal The axis normal to which the gizmo will be able to drag on
+     * @param color The color of the gizmo
+     */
+    constructor(dragPlaneNormal: Vector3, color: Color3 = Color3.Gray(), gizmoLayer: UtilityLayerRenderer = UtilityLayerRenderer.DefaultUtilityLayer, parent: PositionGizmo) {
+        super(gizmoLayer);
+        this._parent = parent;
+        // Create Material
+        var coloredMaterial = new StandardMaterial("", gizmoLayer.utilityLayerScene);

```

Comment:
Yep, any materials created should be disposed in the dispose method. I actually think this is an issue for all the gizmos currently, can you fix that here, I can fix the others.

Reply:
I will add them. :-)

Reply:
@TrevorDev 
Does it makes sense for me to add the dispose in all the others as well? For consistency sake, I am modifying the rest anyway.